src/context/viewer-context.tsx

```diff
@@ -37,6 +38,8 @@ function useAuthedViewer() {
   const previousViewer = React.useRef(viewer)
   const authToken = getAccessTokenFromCookie()
 
+  useTokenSigner()
```

Comment:
I figured the viewer context was as good of place as any to handle a referral token that eventually gets tied to a user's (viewer's) subscription.